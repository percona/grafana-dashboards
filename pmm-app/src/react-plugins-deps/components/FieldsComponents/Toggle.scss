.toggle-field.ant-switch-checked {
  background-color: forestgreen;
}
