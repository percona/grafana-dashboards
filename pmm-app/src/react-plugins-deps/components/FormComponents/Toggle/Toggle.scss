#antd {
  .toggle-field.ant-switch-checked {
    background-color: #92ec94;
  }

  .ant-switch::after {
    background-color: white;
  }
}
