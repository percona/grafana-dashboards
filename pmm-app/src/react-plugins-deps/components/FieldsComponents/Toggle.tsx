import { Switch } from 'antd';
import React from 'react';
import { useField } from 'react-final-form-hooks';
import './Toggle.scss';
export const ToggleField = ({ name, form }) => {
  // @ts-ignore
  const field = useField(name, form);
  return <Switch {...field.input} defaultChecked className={'toggle-field'} />;
};
