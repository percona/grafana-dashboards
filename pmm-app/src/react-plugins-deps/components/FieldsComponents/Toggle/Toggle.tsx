import { Switch } from 'antd';
import React from 'react';
import { useField } from 'react-final-form-hooks';
import './Toggle.scss';
export const ToggleField = ({ name, form }) => {
  // @ts-ignore
  const { input } = useField(name, form);
  return <Switch onChange={input.onChange} checked={Boolean(input.value)} defaultChecked className={'toggle-field'} />;
};
