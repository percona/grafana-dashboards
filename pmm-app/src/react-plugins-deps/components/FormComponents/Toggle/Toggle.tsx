import { Switch } from 'antd';
import React from 'react';
import { useField } from 'react-final-form-hooks';
import './Toggle.scss';

interface ToggleInterface {
  name: string;
  form: any;
  disabled?: boolean;
}
export const ToggleField = ({ name, form, disabled }: ToggleInterface) => {
  // @ts-ignore
  const { input } = useField(name, form);
  return <Switch onChange={input.onChange} checked={Boolean(input.value)} defaultChecked className={'toggle-field'} disabled={disabled} />;
};
