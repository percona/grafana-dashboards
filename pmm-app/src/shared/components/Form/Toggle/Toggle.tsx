import { Switch } from 'antd';
import React from 'react';
import { useField } from 'react-final-form';
import './Toggle.scss';

interface ToggleInterface {
  name: string;
  disabled?: boolean;
}

export const ToggleField = ({ name, disabled }: ToggleInterface) => {
  const { input } = useField(name);

  return (
    <Switch
      onChange={input.onChange}
      checked={Boolean(input.value)}
      className="toggle-field"
      disabled={disabled}
    />
  );
};
