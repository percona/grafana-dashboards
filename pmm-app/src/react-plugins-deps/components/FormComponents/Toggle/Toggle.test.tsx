// Just a stub test
import { ToggleField } from './Toggle';
import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from 'react-final-form';

describe('Toggle field test', () => {
  it('Toggle renders correct without props', () => {
    const component = renderer.create(
      <Form onSubmit={() => null} render={() => <ToggleField name="test_field" />} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
