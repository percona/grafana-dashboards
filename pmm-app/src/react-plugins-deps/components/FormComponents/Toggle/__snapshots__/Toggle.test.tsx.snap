// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle field Toggle renders without props 1`] = `
<ReactFinalForm
  onSubmit={[MockFunction]}
  render={[Function]}
>
  <Component
    name="testField"
  >
    <Switch
      checked={false}
      className="toggle-field"
      onChange={[Function]}
    >
      <Wave
        insertExtraNode={true}
      >
        <Switch
          checked={false}
          checkedChildren={null}
          className="toggle-field"
          defaultChecked={false}
          loadingIcon={null}
          onChange={[Function]}
          prefixCls="ant-switch"
          unCheckedChildren={null}
        >
          <button
            aria-checked={false}
            checked={false}
            className="toggle-field ant-switch"
            defaultChecked={false}
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            role="switch"
            type="button"
          >
            <span
              className="ant-switch-inner"
            />
          </button>
        </Switch>
      </Wave>
    </Switch>
  </Component>
</ReactFinalForm>
`;
